contracts/os/andromeda-kernel/src/ibc.rs

```diff
@@ -131,6 +131,9 @@ pub fn do_ibc_packet_receive(
         CHANNEL_TO_CHAIN.has(deps.storage, channel.as_str()),
         ContractError::Unauthorized {}
     );
+    let chain = CHANNEL_TO_CHAIN.load(deps.storage, channel.as_str())?;
```

Comment:
Doesn't this line make the one above it unnecessary? We could map_err the potential error from loading it.